### PR TITLE
Rework to use spawnLog and errorFinder

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

File: lib/machine/inspection.ts [hunk 1]
```diff
@@ -95,18 +94,21 @@ export function runConfusingDependenciesCheck(): CodeInspection<ProjectReview, N
...
 
-        const result: ExecPromiseResult = await execPromise(
+        const result = await spawnLog(
             "lein",
             ["deps", ":tree"],
             {
                 ...spawnOptions,
                 cwd: project.baseDir,
+                log,
+                errorFinder: (code, signal, l) => l.log.includes("confusing"),
```
Comment (ddgenome) [inline lib/machine/inspection.ts]: Is this enough or do you also have to check for `code !== 0` in the `errorFinder` too?

Comment (cdupuis) [reply lib/machine/inspection.ts]: Don't know, I just converted what was there previously.

Comment (ddgenome) [reply lib/machine/inspection.ts]: I think you might because `execPromise` rejects if `code !== 0` whereas `spawnLog` will just return it.

Comment (cdupuis) [reply lib/machine/inspection.ts]: Right, so if `execPromise` rejects then the previous code wouldn't have created a `ReviewComment` as there is no try/catch in the code. I just tried to keep the behaviour that @slimslenderslacks implemented.

Comment (slimslenderslacks) [reply lib/machine/inspection.ts]: @ddgenome @cdupuis my guess would be that the previous code probably had a bug, but at the same time, we were seeing ReviewComments in cases where the spawned process return non zero.  So I think this means that the current ErrorFinder is fine?

**Pull request merged**